bin/git-authors

```diff
@@ -1,56 +1,69 @@
 #!/usr/bin/env bash
 
```

Comment:
Should the `bin/*` files be modified when we add the tests? And it seems if we modify multiple things, such as the code style and the logic, it is prone to error. Maybe modifications to `bin/*` files could be added in separate PRs, if they are needed at all?

Reply:
Yes, it should be, but there is a change that one more option was added, so I refactor it a lot. And I follow the TDD, unitcase first then to refactor. That is the result here and why i want to do this.

I will recover all of this except something fixed.

Reply:
Done

Reply:
If PRs are squash-merged, then I think the rest of the (large) diff can make it difficult to pinpoint errors or bisect. We don't _know_ if the tests passed before your changes. I think itmight be good to squashing rebase this commit sequence into two commits: one for the `bin/*` code changes, and one for everything else including the test suite (when everything is finally ready). Wonder what @spacewander has to say about this